### ISSUE TITLE
Document does not show function return type which use keyword impl 

### ISSUE THREAD

**Issue by ZhouHansen** [rust-lang/rust#55034]
My function is:
```rust
fn batcher() -> impl Fn(usize) -> () {
// ...
}
```
I run `cargo doc --open`, the document of this function lacks of return type:
```rust
pub fn batcher() -> 
```

cargo version: cargo 1.29.0-nightly

**Comment by GuillaumeGomez** [comment]
Shouldn't it be fixed by #53541? 😮

**Comment by ehuss** [comment]
The fix is not available until 1.30 (beta).
